Userland/Libraries/LibGfx/PortableImageLoaderCommon.h

```diff
@@ -30,158 +30,146 @@ static constexpr Color adjust_color(u16 max_val, Color color)
     return color;
 }
 
-template<typename TValue>
-static bool read_number(Streamer& streamer, TValue* value)
+static inline ErrorOr<u16> read_number(SeekableStream& stream)
 {
-    u8 byte {};
     StringBuilder sb {};
+    u8 byte {};
 
-    while (streamer.read(byte)) {
+    while ((byte = TRY(stream.template read_value<u8>()))) {
         if (byte == ' ' || byte == '\t' || byte == '\n' || byte == '\r') {
-            streamer.step_back();
+            stream.seek(-1, SeekMode::FromCurrentPosition).release_value_but_fixme_should_propagate_errors();
```

Comment:
Do you need the fixme? Can't you just TRY() this too?

Reply:
Oops, looks like a rebase issue.